pkg/agent/endpoints/endpoints_test.go

```diff
@@ -126,6 +129,23 @@ func TestEndpoints(t *testing.T) {
 			name:       "access denied to remote caller",
 			fromRemote: true,
 		},
+		{
+			name: "reflection enabled",
+			do: func(t *testing.T, conn *grpc.ClientConn) {
+				exposedServices := []string{
+					middleware.WorkloadAPIServiceName,
+					middleware.EnvoySDSv3ServiceName,
+					middleware.HealthServiceName,
+					middleware.ServerReflectionServiceName,
+					middleware.ServerReflectionV1AlphaServiceName,
+				}
+				client := grpcreflect.NewClientAuto(context.Background(), conn)
```

Comment:
I would prefer we use the client already provided by grpc-go rather than introducing a new dependency just for this test case: https://pkg.go.dev/google.golang.org/grpc@v1.62.0/reflection/grpc_reflection_v1#NewServerReflectionClient